File: datasets/cpp-keep-code_davinci_001_temp_0.2/HumanEval_19_sort_numbers.cpp
#include<assert.h>
#include<bits/stdc++.h>
// Input is a space-delimited string of numberals from 'zero' to 'nine'.
// Valid choices are 'zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight' and 'nine'.
// Return the string with numbers sorted from smallest to largest
// >>> sort_numbers('three one five')
// 'one three five'
std::string sort_numbers(std::string numbers) {
    std::vector<int> v;
    std::string s;
    for(auto c:numbers){
        if(c==' '){
            v.push_back(atoi(s.c_str()));
            s="";
        }
        else{
            s+=c;
        }
    }
    v.push_back(atoi(s.c_str()));
    std::sort(v.begin(),v.end());
    std::string ret="";
    for(auto i:v){
        ret+=std::to_string(i);
        ret+=" ";
    }
    return ret;

}
int main() {
    auto candidate = sort_numbers;
    assert(candidate(("")) == (""));
    assert(candidate(("three")) == ("three"));
    assert(candidate(("three five nine")) == ("three five nine"));
    assert(candidate(("five zero four seven nine eight")) == ("zero four five seven eight nine"));
    assert(candidate(("six five four three two one zero")) == ("zero one two three four five six"));
}

File: datasets/lua-keep-incoder_6b_temp_0.2-1/HumanEval_19_sort_numbers.lua
-- Input is a space-delimited string of numberals from 'zero' to 'nine'.
-- Valid choices are 'zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight' and 'nine'.
-- Return the string with numbers sorted from smallest to largest
-- >>> sort_numbers('three one five')
-- 'one three five'
local function sort_numbers(numbers)
  local numbers = numbers:gsub('%w+', ''):gsub('%d+', ''):gsub('%w+%d+', ''):gsub('%d+%w+', ''):gsub('%d+%d+%w+', ''):gsub('%d+%d+%w+%d+', ''):gsub('%d+%d+%w+%d+%d+', ''):gsub('%d+%d+%w+%d+%d+%d+', ''):gsub('%d+%d+%w+%d+%d+%d+%d+', ''):gsub('%d+%d+%w+%d+%d+%d+%d+%d+', ''):gsub('%d+%d+%w+%d+%d+%d+%d+%d+%d+%d+', ''):gsub('%d+%d+%w+%d+%d+%d+%d+%d+%d+%d+%d+', ''):gsub('%d+%d+%w+%d+%d+%d+%d+%d+%d+%d+%d+%d+', ''):gsub('%d+%d+%w+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+', ''):gsub('%d+%d+%w+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+', ''):gsub('%d+%d+%w+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+', ''):gsub('%d+%d+%w+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+', ''):gsub('%d+%d+%w+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+', ''):gsub('%d+%d+%w+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+', ''):gsub('%d+%d+%w+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+', ''):gsub('%d+%d+%w+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d+%d

lu = require('luaunit')

function test_humaneval()
local candidate = sort_numbers
    lu.assertEquals(candidate(''), '')
    lu.assertEquals(candidate('three'), 'three')
    lu.assertEquals(candidate('three five nine'), 'three five nine')
    lu.assertEquals(candidate('five zero four seven nine eight'), 'zero four five seven eight nine')
    lu.assertEquals(candidate('six five four three two one zero'), 'zero one two three four five six')
end

os.exit(lu.LuaUnit.run())